packages/compass-saved-aggregations-queries/src/index.spec.tsx

```diff
@@ -8,74 +9,69 @@ import {
 } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { expect } from 'chai';
-import { FavoriteQueryStorage } from '@mongodb-js/my-queries-storage';
-import { PipelineStorage } from '@mongodb-js/my-queries-storage';
-import ConnectedList from './index';
-import * as store from './../stores';
-
-import { queries, pipelines } from '../../test/fixtures';
-import Sinon from 'sinon';
-
-const DATE = new Date('01/01/2020');
+import { queries, pipelines, DATE } from '../test/fixtures';
+import { MyQueriesPlugin } from '.';
 
 describe('AggregationsQueriesList', function () {
-  let sandbox: Sinon.SinonSandbox;
-  let favoriteQueryStorageMock: Sinon.SinonMock;
-  let pipelineStorageMock: Sinon.SinonMock;
-
-  beforeEach(function () {
-    sandbox = Sinon.createSandbox();
-    favoriteQueryStorageMock = sandbox.mock(FavoriteQueryStorage.prototype);
-    pipelineStorageMock = sandbox.mock(PipelineStorage.prototype);
-
-    sandbox.stub(store, 'queryStorage').returns(favoriteQueryStorageMock);
-    sandbox.stub(store, 'pipelineStorage').returns(pipelineStorageMock);
+  const sandbox = Sinon.createSandbox();
+  const dataService = {} as any;
+  const instance = {} as any;
+  const queryStorage = {
+    loadAll: sandbox.stub().resolves([]),
+    updateAttributes: sandbox.stub().resolves({}),
+  };
+  const pipelineStorage = {
+    loadAll: sandbox.stub().resolves([]),
+    updateAttributes: sandbox.stub().resolves({}),
+  };
+
+  const Plugin = MyQueriesPlugin.withMockServices({
+    dataService,
+    instance,
+    queryStorage,
+    pipelineStorage,
   });
 
   afterEach(function () {
-    sandbox.restore();
+    sandbox.resetHistory();
```

Comment:
just a question, any particular reason for this approach here vs creating the sandbox in a beforeEach hook? This obviously works perfectly fine, I'm just a bit surprised (didn't even know that you could do this with sinon :))

Reply:
Just my preference generally for how I set this up, I'm happy to change if you feel like we should be more consistent here. The only difference between this and what you describe is that I can define the mock outside the before hook and don't need an extra `let` definition to be able to access it inside the spec.

Now that I think about it though in this particular case because other tests re-define returned results, it might be better to setup and reset it before and after each test, usually I do this in cases where I only have the same default value everywhere.